docs/examples/dcgan.md

```diff
@@ -0,0 +1,149 @@
+
+```bash
+floyd run --gpu --env pytorch --data <REPLACE_WITH_JOB_OUTPUT_NAME>:model "python generate.py --netG /model/<REPLACE_WITH_MODEL_CHECKPOINT_PATH> --ngpu 1 --cuda"
+
+# Provide a serialized Zvector
+floyd run --gpu --env pytorch --data <REPLACE_WITH_JOB_OUTPUT_NAME>:model "python generate.py --netG /model/<REPLACE_WITH_MODEL_CHECKPOINT_PATH> --Zvector /model/<REPLACE_WITH_SERIALIZED_Z_VECTOR_PATH> --ngpu 1 --cuda"
+```
+
+You can track the status of the run with the status or logs command.
+
+```bash
+$ floyd status <JOB_NAME>
+$ floyd logs <JOB_NAME> -t
+```
+
+
+## Improving your model
+
+You may notice that the output does not look great. In fact, the algorithm have not yet learned how to correctly represent a face.
+That is because we ran the training for a small number of iterations. To train a fully working model, try the training step again, this time by setting the flag `--niter` to a large value, such as 300. In general, about 300/500 epochs(or even more, much more!) are necessary to have an accetable model. (Note: This takes a few hours to run on the GPU instance!).
+Keep in mind that *all the class of generative networks are not neither stable nor production ready*, this is a exciting field of research and everyone can contribute with new ideas.
+
+## Evaluate pre-trained models
+
+If you want to try out a pre-trained model, FloydHub has a public dataset model with the checkpoints(300 epochs training) for
+this. You can mount it with `--data`:
+[redeipirati/datasets/dcgan-300-epochs-models/1](https://www.floydhub.com/redeipirati/datasets/dcgan-300-epochs-models/1)
+.
+
+
+```bash
+floyd run --gpu --env pytorch --data redeipirati/datasets/dcgan-300-epochs-models/1:/model "python generate.py --netG /model/netG_epoch_299.pth --ngpu 1 --cuda"
+```
+
+This model should perform better compared to the previous one. You can also provide the `--Zvector` parameter to explore the latent Z vector landscape. We have also provided to you the zvector used for evaluating our model in the attached dataset:
+
+```bash
+floyd run --gpu --env pytorch --data redeipirati/datasets/dcgan-300-epochs-models/1:/model "python generate.py --netG /model/netG_epoch_299.pth --Zvector /model/zvector.pth --ngpu 1 --cuda"
+```
+
+
+## Serve model through REST API
+
+FloydHub supports seving mode for demo and testing purpose. Before serving your model through REST API,
+you need to create a `floyd_requirements.txt` and declare the flask requirement in it. If you run a job
+with `--mode serve` flag, FloydHub will run the `app.py` file in your project
+and attach it to a dynamic service endpoint:
+
+
+```bash
+floyd run --gpu --mode serve --env pytorch --data <REPLACE_WITH_JOB_OUTPUT_NAME>:/model
+```
+
+Note: If you want to serve the model on a CPU istance, comment this [line](https://github.com/ReDeiPirati/dcgan/blob/master/app.py#L65) and decomment [this other one](https://github.com/ReDeiPirati/dcgan/blob/master/app.py#L67).
```

Comment:
Suggestion, can we use `is_available` function from http://pytorch.org/docs/master/cuda.html to determine if GPU is available? Then we don't need to manually change the code.

The line number link here can easily go stale if you make any change to the source file.

Reply:
Yes, absolutely!! It was a terrible solution and I've forgotten to improve it. Thanks @houqp for the fix!